src/averageAges.js

```diff
@@ -55,7 +75,23 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const findMom = onlyWithSon
+    ? people
+      .filter(person => people.some(mother => mother.name === person.mother)
+       && person.sex === 'm')
+    : people
+      .filter(person => people.some(mother => mother.name === person.mother));
```

Comment:
you repeat this 2 times, you can make it cleaner if you put this to the function helper like
```suggestion
  const findMom = onlyWithSon
    ? people.filter(person => hasMother(people, person.mother) && person.sex === 'm'
    : people.filter(person => hasMother(people, person.mother));
```

OR
```suggestion
  const findMom = people.filter(person =>  onlyWithSon 
    ? hasMother(people, person.mother) && person.sex === 'm'
    : hasMother(people, person.mother));
```